[BUG] (type-safe-api) src/apis/DefaultApiClientProvider.tsx(37,36): error TS2503: Cannot find namespace 'JSX'

### Describe the bug

Generated typescript hooks library fails to compile

```
src/apis/DefaultApiClientProvider.tsx(37,36): error TS2503: Cannot find namespace 'JSX'
```

### Expected Behavior

Should compile

### Current Behavior

Fails to compile

### Reproduction Steps

Follow getting started guide https://aws.github.io/aws-pdk/getting_started/your_first_aws_pdk_project.html#creating-our-aws-infrastructure

### Possible Solution

React 19 was released recently and the hooks library doesn't pin the version. React 19 removes the global JSX namespace: https://react.dev/blog/2024/04/25/react-19-upgrade-guide#the-jsx-namespace-in-typescript

We can either pin to react 18 or investigate if `React.JSX` works in both 18 and 19.

### Additional Information/Context

Current workaround in pin react at version 18 in your `.projenrc.ts`:

```ts
api.library.typescriptReactQueryHooks!.addPeerDeps("react@^18");
api.library.typescriptReactQueryHooks!.addDevDeps("@types/react@^18");
```

### PDK version used

0.25.14

### What languages are you seeing this issue on?

Typescript

### Environment details (OS name and version, etc.)

osx

--- a/packages/cloudscape-react-ts-website/samples/cloudscape-react-ts-website/src/layouts/App/index.tsx.mustache
+++ b/packages/cloudscape-react-ts-website/samples/cloudscape-react-ts-website/src/layouts/App/index.tsx.mustache
@@ -78,7 +78,7 @@ const App: React.FC = () => {
           splitPanelSize: undefined,
           splitPanelPreferences: undefined,
         });
-        navigate(e.detail.href);
+        void navigate(e.detail.href);
       }
     },
     [navigate],

--- a/packages/cloudscape-react-ts-website/samples/cloudscape-react-ts-website/src/pages/Home/index.tsx.mustache
+++ b/packages/cloudscape-react-ts-website/samples/cloudscape-react-ts-website/src/pages/Home/index.tsx.mustache
@@ -26,7 +26,7 @@ import { useNavigate } from "react-router-dom";
 const Home: React.FC = () => {
   {{#hasApis}}
   const navigate = useNavigate();
-  useEffect(() => navigate("/apiExplorer"), []);
+  useEffect(() => void navigate("/apiExplorer"), []);
   {{/hasApis}}
 
   {{#typeSafeApis.0}}

--- a/packages/type-safe-api/scripts/type-safe-api/generators/typescript-react-query-hooks/templates/clientProvider.ejs
+++ b/packages/type-safe-api/scripts/type-safe-api/generators/typescript-react-query-hooks/templates/clientProvider.ejs
@@ -48,7 +48,7 @@ export const <%- service.className %>ClientProvider = ({
   context = <%- service.className %>DefaultContext,
   <%_ } _%>
   children,
-}: <%- service.className %>ClientProviderProps): JSX.Element => {
+}: <%- service.className %>ClientProviderProps): React.JSX.Element => {
   return (
     <QueryClientProvider client={client}<% if (!metadata.queryV5) { %> context={context}<% } %>>
       <<%- service.className %>ClientContext.Provider value={apiClient}>

